frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/common/PortletUtils.java

```diff
@@ -143,48 +146,61 @@ else if (field == Project._Fields.VISBILITY)
     }
 
 
-    public static void updateAttachmentsFromRequest(PortletRequest request, Set<Attachment> attachments) {
-        if (attachments == null || attachments.size() == 0) return;
+    public static Set<Attachment> updateAttachmentsFromRequest(PortletRequest request, Set<Attachment> documentAttachments) {
+        if (documentAttachments == null) {
+            log.error("UpdateAttachments called with null documentAttachments.");
+            return null;
```

Comment:
if you assign `documentAttachments = new HashSet<>();` and work with that instead of returning null, you can spare the checks like `if (!component.isSetAttachments()) component.setAttachments(new HashSet<Attachment>());` at the three places where this method is called.
That's three lines of code savings and a more robust method. Looks like win to me.